### PR TITLE
Validate submodel reference in JSON Schema

### PR DESCRIPTION
**Pull request by mhrimaz**
Check submodel references to be only `ModelReference<SubmodelReference>`

### CODE REVIEW

**Comment by mristin** [comment]
@mhrimaz Thanks for the PR!

@sebbader-sap @s-heppner maybe we should collect all these patches and re-write them  as Python code so that they can be re-applied to new versions as well?

**Comment by mristin** [comment]
@sebbader-sap the constraint has been in aas-core-meta since forever, but we can not transpile it to JSON schema. At least, I don't know how.

File: schemas/json/aas.json [hunk 1]
```diff
@@ -1207,6 +1224,34 @@
...
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Reference"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "type": {
+              "const": "ModelReference"
```
Comment (BirgitBoss) [inline schemas/json/aas.json:1237]:
```suggestion
              "const": "Submodel"
```

Comment (mhrimaz) [reply schemas/json/aas.json:1237]: Thanks for the review @BirgitBoss.
The schema that I provided only enforces that the reference should be `ModelReference` and not `ExternalReference` with only one `Key`. So for enforcing that the key should refer to a `Submodel` then, we should further extend it.

In any case the fix should be in the aas-core schema generator.

**Comment by s-heppner** [comment]
We truly thank you for your input, @mhrimaz!

However, we sadly cannot allow changing the schema files by hand, since they are generated using [aas-core-works](https://github.com/aas-core-works). 
That particular constraint is implemented in [aas-core-meta](https://github.com/aas-core-works/aas-core-meta) (meaning it appears for example in the generated SDKs), however cannot be transpiled into the JSON schema with reasonable effort, so that we decided not to implement that particular generation function. 
If you're willing to, you're more than welcome to contribute to the schema generation functionality at [aas-core-codegen](https://github.com/aas-core-works/aas-core-codegen). 

@mristin We decided against the semantic patching approach you mentioned due to not wanting to introduce another layer of code that needs to be maintained.

**Comment by mristin** [comment]
I'll close the pull request for now, but feel free to re-open it in the future.

**Pull request closed without merging**